parser/src/parsing.rs

```diff
@@ -106,38 +106,50 @@ impl Parser {
         let mut term = Term::Factor(self.factor().await?);
 
         loop {
-            let next = self.reader.peek().await;
-            match next {
-                Token::Asterisk => {
+            let next = self.reader.peek_range(0..=1).await;
+            match next[..] {
+                [Token::Asterisk, _] => {
                     self.reader.skip().await;
                     let rhs = self.factor().await?;
-                    term = Term::Multiply(Box::new(term), rhs);
+                    term =
+                        Term::Multiply(MulType::Asterisk, Box::new(term), rhs);
                 }
-                Token::Slash => {
+                [Token::Backslash, Token::Identifier(ident)]
+                    if ident == "cdot"
+                        || ident == "cdotp"
+                        || ident == "times" =>
+                {
+                    let mul_type = match ident.as_str() {
+                        "cdot" | "cdotp" => MulType::Cdot,
+                        "times" => MulType::Times,
+                        _ => unreachable!(),
+                    };
```

Comment:
feels like this is double work but i dont have a better sugestion

Reply:
I don't think there is a better alternative, since splitting them up into separate cases in the match-statement would introduce more duplicate code.

Reply:
What if we like matched on it and if it was not we did the rest of the code and if not we returned?
example:
```
match ident {
    "cdot"|"cdotp"=>{return Self::Mul(MulType::Cdot,...)},
    "times"=>{return Self::Mul(MulType::Cdot,...)}
    ident=>{
        // noraml \ then ident code here
    }
}
```
kinda like this?

Reply:
If we get an ident that isn't cdot(p) or times, it could be \end in which case we should break out of the loop, or it could be a factor in which case we need to do other logic.

Unless Rust has some continue keyword for match-statements to continue pattern matching I think that would cause more duplicate code.

Reply:
I think it can be done but as that is a micro optimization Il just make a pr if I get to it

Reply:
We could use the ```rust
match ident {
    "cdot"|"cdotp" if *condition* => {...}
    ...
}``` maybe?